src/mo_frontend/typing.ml

```diff
@@ -1150,7 +1150,18 @@ and infer_exp'' env exp : T.typ =
         { env with async = C.NullCap; in_actor = true }
       else env
     in
-    infer_obj env' obj_sort.it dec_fields exp.at
+    let t = infer_obj env' obj_sort.it dec_fields exp.at in
+    begin match typ_opt with
```

Comment:
I wonder if you only want to do the check in the second pass, something like:

```suggestion
   if not (env.pre) then begin match typ_opt with
```
Otherwise you might get repeated local errors on certain occasions. (The type checker has become a mystery to me, tbh).

Reply:
a81acec6a